component/exporter.go

```diff
@@ -111,3 +117,17 @@ type ExporterFactory interface {
 	CreateMetricsExporter(ctx context.Context, params ExporterCreateParams,
 		cfg configmodels.Exporter) (MetricsExporter, error)
 }
+
+// LogExporterFactory can create a LogExporter.
+type LogExporterFactory interface {
```

Comment:
Can we break down `ExporterFactory` later as well? In that case we probably don't have to create empty `CreateTraceExporter` or `CreateMetricsExporter` raising `ErrDataTypeIsNotSupported` for exporter that works with metrics or traces only.

Reply:
Yes. @bogdandrutu this is also what you asked for :-)
This can be done in a future PR.